src/coord_seq.rs

```diff
@@ -149,6 +149,93 @@ impl<'a> CoordSeq<'a> {
         }
     }
 
+    #[cfg(any(feature = "v3_10_0", feature = "v3_11_0"))]
+    pub fn new_from_buffer<T: AsRef<[f64]>>(
+        data: &[f64],
+        size: u32,
+        dims: u32,
+        has_z: bool,
+        has_m: bool,
+    ) -> GResult<CoordSeq<'a>> {
+        let data_ptr = &data[0] as *const f64;
+        assert_eq!(
+            data.len(),
+            (size * dims) as usize,
+            "Incorrect buffer length"
+        );
+
+        match ContextHandle::init_e(Some("CoordSeq::new_from_buffer")) {
+            Ok(context_handle) => unsafe {
+                let ptr = GEOSCoordSeq_copyFromBuffer_r(
+                    context_handle.as_raw(),
+                    data_ptr,
+                    size as _,
+                    has_z as _,
+                    has_m as _,
+                );
+                CoordSeq::new_from_raw(
+                    ptr,
+                    Arc::new(context_handle),
+                    size,
+                    dims.into(),
+                    "new_from_buffer",
+                )
+            },
+            Err(e) => Err(e),
+        }
```

Comment:
I think this will work too:

```suggestion
        let context_handle = ContextHandle::init_e(Some("CoordSeq::new_from_buffer"))?;
        let ptr = GEOSCoordSeq_copyFromBuffer_r(
                context_handle.as_raw(),
                data_ptr,
                size as _,
                has_z as _,
                has_m as _,
            );
            CoordSeq::new_from_raw(
                ptr,
                Arc::new(context_handle),
                size,
                dims.into(),
                "new_from_buffer",
            )
```

Reply:
Makes sense; I was copying from `new()` which used match

Reply:
Some parts of this crate are **old**. So nothing surprising. ;)